pkgs/tools/package-management/dpkg/default.nix

```diff
@@ -1,20 +1,21 @@
-{ lib, stdenv, fetchurl, perl, zlib, bzip2, xz, zstd
-, makeWrapper, coreutils, autoreconfHook, pkg-config
+{ lib, stdenv, fetchurl, perl, gnutar, zlib, bzip2, xz, zstd
+, libmd, makeWrapper, coreutils, autoreconfHook, pkg-config
 }:
 
 stdenv.mkDerivation rec {
   pname = "dpkg";
-  version = "1.21.1ubuntu2.1";
+  version = "1.21.19ubuntu2";
 
   src = fetchurl {
     url = "mirror://ubuntu/pool/main/d/dpkg/dpkg_${version}.tar.xz";
```

Comment:
Is there a better download link available? otherwise please use web.archive.org.

Reply:
Not sure if there is a better link, I will check. If cannot find something more stable I will add a web.archive.org link as fallback.

Reply:
The Git repository is now used as the source, it should avoid the disappearing source tarball issue.

Reply:
Maybe it would make more sense to just pull it directly from Debian's GitLab (https://salsa.debian.org/dpkg-team/dpkg), and if we need the Ubuntu patches, apply them here directly?

Reply:
We could but I'm not sure what we would gain from it. Applying the patches ourselves means more work for us. Dealing with the upgrades is already tedious as demonstrated by this PR, adding patch management on top of it will not help :smile: